File: myshop/src/main/java/study/spring/myshop/common/model/Money.java
package study.spring.myshop.common.model;

import lombok.Getter;

@Getter
public class Money {

    private int value;

    public Money(int value) {
        this.value = value;
    }

    public Money multiply(int multiplier) {
        return new Money(value * multiplier);
    }

    @Override
    public String toString() {
        return Integer.toString(value);
    }
}

File: 2nd/myshop/src/main/kotlin/study/ddd/myshop/common/model/Money.kt
package study.ddd.myshop.common.model

import javax.persistence.Access
import javax.persistence.AccessType
import javax.persistence.Column
import javax.persistence.Embeddable

@Embeddable
data class Money(
    @Column(name = "money_value")
    @Access(AccessType.FIELD)
    val value: Int
) {

    fun add (money: Money): Money = Money(value + money.value)

    fun multiply(multiplier: Int): Money = Money(value * multiplier)
}